File: 03-encapsulamiento/biblioteca/biblioteca.py
"""Class Biblioteca"""

from libro import Libro


class Biblioteca:
    def __init__(self):
        self.__libros: list[Libro] = []

    def cuantos_libros(self):
        """Devuelve la cantidad de libros en la biblioteca"""
        return len(self.__libros)

    def agregar_libro(self, titulo, autor, genero, paginas):
        """Agrega un Libro a la biblioteca"""
        self.__libros.append(Libro(titulo, autor, genero, paginas))

    def libro_en_la_posicion(self, posicion):
        """Devuelve el título de un libro a partir de la posición."""
        return self.__libros[posicion - 1].obtener_titulo()

    def libro_con_mas_paginas(self) -> Libro:
        """Devuelva el libro con más cantidad de páginas."""
        pos_max = 0
        max_pag = self.__libros[0].obtener_paginas()
        for i in range(1, len(self.__libros)):
            if self.__libros[i].obtener_paginas() > max_pag:
                pos_max = i
                max_pag = self.__libros[i].obtener_paginas()
        return self.__libros[pos_max]


def main():
    pass


if __name__ == "__main__":
    main()

File: workspace-2025-1c/Biblioteca/src/biblioteca/Biblioteca.java
package biblioteca;

import java.util.ArrayList;

public class Biblioteca {

	private ArrayList<Libro> libros;

	public Biblioteca() {
		this.libros = new ArrayList<Libro>();
	}

	public void agregarLibro(String titulo, String autor, Genero genero, int paginas) {
		libros.add(new Libro(titulo, autor, genero, paginas));
	}

	public int cuantosLibros() {
		return this.libros.size();
	}

	public String libroEnLaPosicion(int pos) {
		return this.libros.get(pos - 1).getTitulo();
	}

	public Libro libroConMasPaginas() {
		int posMax = 0;
		int maxPag = this.libros.get(posMax).getCantidadDePaginas();

		for (int i = 1; i < this.libros.size(); i++) {
			if (this.libros.get(i).getCantidadDePaginas() > maxPag) {
				posMax = i;
				maxPag = this.libros.get(i).getCantidadDePaginas();
			}

		}
		return this.libros.get(posMax);

	}

	public void librosPorGeneroLiterario() {
		Genero generos[] = Genero.values();
		int cuantos[] = new int[generos.length];
		for (Libro cadaLibro : libros) {
			int i = cadaLibro.getGenero().ordinal();
			cuantos[i]++;
		}
		for (int i = 0; i < generos.length; i++)
			System.out.println(generos[i] + " " + cuantos[i]);

	}
}
